cmake/install/ConfigurationTypes.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+include_guard(GLOBAL)
+
+
+# In SDK builds CMAKE_CONFIGURATION_TYPES will be filled by entries generated per configuration build.
+# At install time we generate `cmake/ConfigurationTypes_<config>.cmake` files that append the configuration
+# to CMAKE_CONFIGURATION_TYPES
+set(CMAKE_CONFIGURATION_TYPES "" CACHE STRING "" FORCE)
+
+# For the SDK case, we want to only define the confiuguration types that have been added to the SDK
+file(GLOB configuration_type_files "cmake/ConfigurationTypes_*.cmake")
+foreach(configuration_type_file ${configuration_type_files})
+    include(${configuration_type_file})
+endforeach()
+ly_set(CMAKE_CONFIGURATION_TYPES ${CMAKE_CONFIGURATION_TYPES}) # propagate to parent
```

Comment:
Does this work if we configure twice on an installed SDK?

Wouldn't the local CMAKE_CONFIGURATION_TYPES variable be initialized to cache value of CMAKE_CONFIGURATION_TYPES the second time when running the logic in the `ConfigurationTypes_${CMAKE_INSTALL_CONFIG_NAME}.cmake` files?

Reply:
it doesnt, but will clear for safety

Reply:
Ah that makes sense. It makes sure the `$CACHE{CMAKE_CONFIGURATION_TYPES}` is empty